README.md

```diff
@@ -169,6 +169,107 @@ package doesn't currently support the functionality that you need.
 
 
 
+
+# A Worked Example
+
+
+
+We use a model involving a Gaussian process (GP) -- you don't need to know anything about
+Gaussian processes other than
+1. they are a class of probabilistic model which can be used for regression (amongst other things).
+2. they have some tunable free-parameters that are usually tuned using an iterative
+optimisation algorithm -- typically a variant on gradient descent.
+Ths is representative of a large number of models in ML / statistics / optimisation.
+
+This example can be copy+pasted into a REPL session.
+
+```julia
+# Install some packages.
+using Pkg
+Pkg.add("ParameterHandling")
+Pkg.add("Optim")
+Pkg.add("Zygote")
+Pkg.add("AbstractGPs")
+
+using ParameterHandling # load up this package.
+using Optim # generic optimisation
+using Zygote # algorithmic differentiation
+using AbstractGPs # package containing the models we'll be working with
+
+# Declare a NamedTuple containing an initial guess at parameters.
+raw_initial_params = (
+    k1 = (
+        var=positive(0.9),
+        precision=positive(1.0),
+    ),
+    k2 = (
+        var=positive(0.1),
+        precision=positive(0.3),
+    ),
+    noise_var=positive(0.2),
+)
+
+# Using ParameterHandling.flatten, we can obtain both a Vector{Float64} representation of
+# these parameters, and a mapping from that vector back to the original parameters:
+flat_initial_params, unflatten = ParameterHandling.flatten(raw_initial_params)
+
+# ParameterHandling.value strips out all of the Positive types in initial_params,
+# returning a plain named tuple of named tuples and Float64s.
+initial_params = ParameterHandling.value(raw_initial_params)
+
+# We define unpack to map directly from the flat Vector{Float64} representation to a
+# the NamedTuple representation with all the Positive types removed.
+unpack = ParameterHandling.value ∘ unflatten
+
+# GP-specific functionality. Don't worry about the details, just
+# note the use of the structured representation of the parameters.
+function build_gp(params)
```

Comment:
Add type constraint just for reader's edification?
Either on params arg,
or inside the function on k1 and k2 or both